url542ae48576_tonyheadford_m68k_tgz-pJ8-m68k_MonitorJ8/src/m68k/Monitor.java

```diff
@@ -0,0 +1,966 @@
+//    o  Neither the name of the M68k Project nor the names of its contributors may be used to endorse or promote
+//       products derived from this software without specific prior written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+//  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+//  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+//  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+//  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+//  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+*/
+public class Monitor implements Runnable
+{
+	private final Cpu cpu;
+	private final AddressSpace memory;
+	private boolean running;
+	private StringBuilder buffer;
+	private BufferedReader reader;
+	private PrintWriter writer;
+	private boolean showBytes;
+	private boolean autoRegs;
+	private ArrayList<Integer> breakpoints;
+
+	public Monitor(Cpu cpu, AddressSpace memory)
+	{
+		this.cpu = cpu;
+		this.memory = memory;
+		buffer = new StringBuilder(128);
+		showBytes = false;
+		autoRegs = false;
+		breakpoints = new ArrayList<Integer>();
+	}
+
+	public static void main(String[] args)
+	{
+		//TODO: commandline args could be expanded in future to include CPU type etc.
+
+		int mem_size = 512;	// 512kb of memory default
+
+		if(args.length == 1)
+		{
+			// initial memory size
+			try
+			{
+				mem_size = Integer.parseInt(args[0]);
+			}
+			catch(NumberFormatException e)
+			{
+				System.err.println("Invalid number: " + args[0]);
+				System.out.println("Usage: m68k.Monitor [memory size Kb]");
+				System.exit(-1);
+			}
+		}
+
+		System.out.println("m68k Monitor v0.1 - Copyright 2008-2010 Tony Headford");
+
+		AddressSpace memory = new MemorySpace(mem_size);
+		Cpu cpu = new MC68000();
+		cpu.setAddressSpace(memory);
+		cpu.reset();	//init cpu
+
+		Monitor monitor = new Monitor(cpu,memory);
+		monitor.run();
+	}
+
+	public void run()
+	{
+		Console con = System.console();
+		if(con != null)
+		{
+			writer = con.writer();
+			reader = new BufferedReader(con.reader());
+		}
+		else
+		{
+			writer = new PrintWriter(System.out);
+			reader = new BufferedReader(new InputStreamReader(System.in));
+		}
+
+		running = true;
+		while(running)
+		{
+			try
+			{
+				writer.print("> ");
+				writer.flush();
+				handleCommand(reader.readLine());
+
+				if(autoRegs)
+				{
+					dumpInfo();
+				}
+			}
+			catch(IOException e)
+			{
+				e.printStackTrace();
+			}
+		}
+	}
+
+	protected void handleCommand(String line)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 24. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.